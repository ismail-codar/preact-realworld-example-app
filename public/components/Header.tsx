import Link from './Link';
import useStore from '../store';

export default function Header() {
	const { isAuthenticated, user } = useStore(state => ({
		isAuthenticated: state.isAuthenticated,
		user: state.user
	}));

	return (
		<nav class="navbar navbar-light">
			<div class="container">
				<a class="navbar-brand" href="/">
					conduit
				</a>
				<ul class="nav navbar-nav pull-xs-right">
					<li class="nav-item">
						<Link href="/">
							Home
						</Link>
					</li>
					{isAuthenticated ? (
						<>
							<li class="nav-item">
								<Link href="/editor">
									<i class="ion-compose" /> New Article
								</Link>
							</li>
							<li class="nav-item">
								<Link href="/settings">
									<i class="ion-gear-a" /> Settings
								</Link>
							</li>
							<li class="nav-item">
								<Link
									matcher={(url) => /^\/@.*/g.test(url)}
									href={`/@${user?.username}`}
								>
									{user?.username}
								</Link>
							</li>
						</>
					) : (
						<>
							<li class="nav-item">
								<Link href="/login">
									Sign in
								</Link>
							</li>
							<li class="nav-item">
								<Link href="/register">
									Sign up
								</Link>
							</li>
						</>
					)}
				</ul>
			</div>
		</nav>
	);
}
