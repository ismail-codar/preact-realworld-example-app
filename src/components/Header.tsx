import { Fragment, FunctionalComponent, h } from 'preact'
import { Link } from 'preact-router'
import { useRootState } from '../store'

const Header: FunctionalComponent = () => {
  const [{ user }] = useRootState()

  return <nav className="navbar navbar-light">
    <div className="container">
      <a className="navbar-brand" href="/">conduit</a>
      <ul className="nav navbar-nav pull-xs-right">
        <li className="nav-item">
          <Link className="nav-link" activeClassName="active" href="/">Home</Link>
        </li>
        {
          user ? (
            <Fragment>
              <li className="nav-item">
                <Link className="nav-link" href="/article/create">
                  <i className="ion-compose" /> New Post
                </Link>
              </li>
              <li className="nav-item">
                <Link className="nav-link" activeClassName="active" href="/settings">
                  <i className="ion-gear-a" /> Settings
                </Link>
              </li>
              <li className="nav-item">
                <Link className="nav-link" activeClassName="active" href={`/@${user?.username}`}>
                  {user?.username}
                </Link>
              </li>
            </Fragment>
          ) : (
            <Fragment>
              <li className="nav-item">
                <Link className="nav-link" activeClassName="active" href="/register">Sign up</Link>
              </li>
              <li className="nav-item">
                <Link className="nav-link" activeClassName="active" href="/login">Sign in</Link>
              </li>
            </Fragment>
          )
        }
      </ul>
    </div>
  </nav>
}

export default Header
