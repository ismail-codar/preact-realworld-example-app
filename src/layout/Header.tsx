import { Component, Fragment, h } from 'preact'
import { Link } from 'preact-router'
import { useContext } from 'preact/hooks'
import RootStateContext from '../stores/globalContext'

export default class Header extends Component {
  render() {
    const rootState = useContext(RootStateContext)

    return <nav className="navbar navbar-light">
      <div className="container">
        <a className="navbar-brand" href="/">conduit</a>
        <ul className="nav navbar-nav pull-xs-right">
          <li className="nav-item">
            <Link className="nav-link" activeClassName="active" href="/">Home</Link>
          </li>
          {
            rootState.user ? (
              <Fragment>
                <li className="nav-item">
                  <a className="nav-link" href="/settings">
                    <i className="ion-compose" /> New Post
                  </a>
                </li>
                <li className="nav-item">
                  <Link className="nav-link" activeClassName="active" href="/settings">
                    <i className="ion-gear-a" /> Settings
                  </Link>
                </li>
              </Fragment>
            ) : (
              <Fragment>
                <li className="nav-item">
                  <Link className="nav-link" activeClassName="active" href="/register">Sign up</Link>
                </li>
                <li className="nav-item">
                  <Link className="nav-link" activeClassName="active" href="/login">Sign in</Link>
                </li>
              </Fragment>
            )
          }
        </ul>
      </div>
    </nav>
  }
}

