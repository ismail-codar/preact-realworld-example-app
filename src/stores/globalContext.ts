import { createContext } from 'preact'

export const initialRootState: RootState = {
  user: null
}

const RootStateContext = createContext<RootState>(initialRootState)

export default RootStateContext
