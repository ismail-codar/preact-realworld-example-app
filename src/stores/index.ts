import { createContext } from 'preact'
import parseStorageGet from '../utils/parse-storage-get'

export const initialRootState: RootState = {
  user: parseStorageGet('user') || null,
}

export const RootContext = createContext<RootState>(initialRootState)
