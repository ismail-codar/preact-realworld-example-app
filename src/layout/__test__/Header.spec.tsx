import { h } from 'preact'
import { useContext } from 'preact/hooks'
import { initialRootState } from '../../stores/globalContext'
import Header from '../Header'
import render from 'preact-render-to-string'

jest.mock('preact/hooks', () => ({
  useContext: jest.fn(() => initialRootState),
}))

afterEach(() => {
  jest.resetAllMocks()
})

describe('# Header Component', () => {
  it('should hidden post and settings and display logging buttons when not logged', () => {
    (useContext as jest.Mock).mockReturnValue({
      user: null,
    })
    const html = render(<Header />)

    expect(html).toMatchSnapshot()
  })

  it('should display post and settings and hidden logging buttons when logged', () => {
    (useContext as jest.Mock).mockReturnValue({
      user: {},
    })
    const html = render(<Header />)

    expect(html).toMatchSnapshot()
  })
})
