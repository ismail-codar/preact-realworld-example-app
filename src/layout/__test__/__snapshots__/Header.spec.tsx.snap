// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Header Component should display post and settings and hidden logging buttons when logged 1`] = `
<nav class="navbar navbar-light">
  <div class="container">
    <a class="navbar-brand" href="/">conduit</a>
    <ul class="nav navbar-nav pull-xs-right">
      <li class="nav-item">
        <a class="nav-link" activeClassName="active" href="/">Home</a>
      </li>
      <li class="nav-item">
        <a class="nav-link" href="/settings">
          <i class="ion-compose"></i>New Post</a>
      </li>
      <li class="nav-item">
        <a class="nav-link" activeClassName="active" href="/settings">
          <i class="ion-gear-a"></i>Settings</a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`# Header Component should hidden post and settings and display logging buttons when not logged 1`] = `
<nav class="navbar navbar-light">
  <div class="container">
    <a class="navbar-brand" href="/">conduit</a>
    <ul class="nav navbar-nav pull-xs-right">
      <li class="nav-item">
        <a class="nav-link" activeClassName="active" href="/">Home</a>
      </li>
      <li class="nav-item">
        <a class="nav-link" activeClassName="active" href="/register">Sign up</a>
      </li>
      <li class="nav-item">
        <a class="nav-link" activeClassName="active" href="/login">Sign in</a>
      </li>
    </ul>
  </div>
</nav>
`;
